### PR TITLE
feat(app): add cloud-aware author editability to AuthorChip

### PR DESCRIPTION
**Pull request by mldangelo**
## Summary

This PR makes eval authorship follow a clear ownership contract in Promptfoo Cloud:

- Cloud users can claim an eval only while it is unassigned.
- Once claimed, the author is immutable; the UI and server both enforce that rule.
- Self-hosted users keep the existing ability to set or clear an author freely.

## Design and Security Notes

The server is authoritative for Cloud ownership. In addition to enforcing claim-only
updates on `PATCH /api/eval/:id/author`, this branch prevents manually rewriting or
clearing the locally stored email through `/api/user/email` while Cloud authentication
is enabled. Otherwise, a caller could set an arbitrary "current user" email and then
claim an unassigned eval under that identity.

Cloud login/logout remain the supported paths for managing the authenticated identity.
The author chip also applies the claim-only rule defensively even if a parent supplies
an inconsistent `editable` prop.

## Main Synchronization

- Merged current `origin/main` into this branch.
- Resolved the `ResultsView.test.tsx` overlap by retaining the Cloud config mock while
  preserving the newer navigation test setup from `main`.
- No database migrations were needed for this behavior change.

## Validation

- [x] `npm run l` (passes with the pre-existing cognitive-complexity warning in `src/models/eval.ts`)
- [x] `npm run tsc`
- [x] `npm run build`
- [x] Frontend focused tests: 69 passed (`AuthorChip`, `EvalHeader`, `ResultsView`)
- [x] Server/model focused tests: 233 passed (`eval`, validation, user route, smoke, eval model)
- [x] Full app suite: 4,553 tests passed
- [x] Full backend suite: 18,195 tests passed, 10 skipped
- [x] Isolated live CLI/server E2E using a temporary `PROMPTFOO_CONFIG_DIR` and cache:
  created an eval with the local build, confirmed self-hosted author set/clear, then
  confirmed Cloud mode rejects identity rewrite/clear, rejects claiming as another
  identity, accepts the authenticated user's first claim, and rejects later
  author clear/replacement attempts.
- [x] Confirmed temporary SQLite state files were created with mode `0600`.

`npm run f` was attempted, but its changed-file wrapper invokes Prettier without input
when the effective diff contains no Prettier-target file paths. Biome successfully
checked the PR files; the seven additional fix/test files were explicitly formatted
with Biome before the lint, typecheck, test, and build gates above.

### CODE REVIEW

**Comment by ghost** [comment]
**⚠️ Failed to commit 7 tests** @mldangelo

There was an unexpected error when trying to commit the selected tests, go to [Tusk](https://app.usetusk.ai/app/testing-commit-check/05298f63-c191-44e4-a094-ee96c0af2b70?client=e91e3433-8fb4-4e98-b233-bba492f4909c) to retry committing.
      
Please contact [Support](https://app.usetusk.ai/app/testing-commit-check/05298f63-c191-44e4-a094-ee96c0af2b70?client=e91e3433-8fb4-4e98-b233-bba492f4909c&showChatBubble=true) if you have any questions.

**Comment by ghost** [comment]
## ✅ Generated 8 tests - 8 passed (70fc872dbc144fc7772cc1f3f73c505f858aaaef) [View tests ↗](https://app.usetusk.ai/app/testing-commit-check/83892fab-e864-4984-9b83-68d122f18493?client=e91e3433-8fb4-4e98-b233-bba492f4909c)

**⚠️ Failed to commit 8 tests - See [comment](https://github.com/promptfoo/promptfoo/pull/6759#issuecomment-3701636191) for details.**

### Test Summary

- `AuthorChip` - 5 ✅
- `ResultsView` - 3 ✅

### Results

Tusk's tests are all passing and cover the critical paths for cloud-aware author management. The `AuthorChip` tests validate the core business logic: preventing unclaimed evals from being modified when the user isn't logged in, correctly displaying ownership messages before claiming, and handling the non-editable state for your own eval. The `ResultsView` tests ensure the component doesn't crash on empty states and that eval selection navigation works correctly. The test suite exercises both the happy paths (claiming an eval, selecting from recent evals) and defensive edge cases (null user email, missing eval IDs, non-Error rejections), which is solid coverage for a feature that touches auth, cloud configuration, and user workflows.

---
<details><summary>View check history</summary>
<p>

| Commit | Status | Output | Created (UTC) |
|--------|--------|--------|--------|
| 858b40c151c8276823ecd5b4b8b1902ee9b65ce6 | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/11f4fac4-98b8-4f0c-b8dd-b56158dce5fe?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 18, 2025 8:32AM |
| 35fbadfa110063eab839e04ca842164b6f34fe37 | ⏩ Test environment setup failed | [Output](https://app.usetusk.ai/app/testing-commit-check/e9c03665-96af-465d-a8ca-cdf38fb2380e?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 18, 2025 8:43AM |
| bb5cf01e84f300efa385fd96265c45d9ec44eaf2 | ⏩ No tests generated | [Output](https://app.usetusk.ai/app/testing-commit-check/0cd6a94a-47a2-4823-9190-61c0c9343b92?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 18, 2025 9:24AM |
| a00ec26e8323ff0a9a8694c3b53e1951905176e4 | ⏩ Skipped due to new commit on branch | [Output](https://app.usetusk.ai/app/testing-commit-check/4f6a2581-7c28-4632-869d-149009dda8b9?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 19, 2025 5:35PM |
| f21ebb556d5cbc4877b8e7164aaf012277225222 | ❌ Generated 9 tests - 7 passed, 2 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/735258d2-f88c-48f7-a238-7b05ba639432?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 19, 2025 5:40PM |
| 311e92b67721a9e4e0022e5e8042c7f139ca3267 | ❌ Generated 7 tests - 4 passed, 3 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/05298f63-c191-44e4-a094-ee96c0af2b70?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 19, 2025 6:18PM |
| d3a877aa00973e916f65b5c079f5481300b6883a | ❌ Generated 7 tests - 4 passed, 3 failed | [Tests](https://app.usetusk.ai/app/testing-commit-check/9c97bafd-8b0a-4179-b28c-307720a45b4f?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 30, 2025 7:57AM |
| 70fc872dbc144fc7772cc1f3f73c505f858aaaef | ✅ Generated 8 tests - 8 passed | [Tests](https://app.usetusk.ai/app/testing-commit-check/83892fab-e864-4984-9b83-68d122f18493?client=e91e3433-8fb4-4e98-b233-bba492f4909c) | Dec 30, 2025 8:50AM |

</p>
</details>

[View output in GitHub ↗](https://github.com/promptfoo/promptfoo/pull/6759/checks?check_run_id=59140736181)

_Was Tusk helpful? Give feedback by reacting with 👍 or 👎_

**Comment by ghost** [comment]
**⚠️ Successfully committed 5 of 8 selected tests** @mldangelo

There was an unexpected error when trying to commit some tests, go to [Tusk](https://app.usetusk.ai/app/testing-commit-check/83892fab-e864-4984-9b83-68d122f18493?client=e91e3433-8fb4-4e98-b233-bba492f4909c) to see which tests were committed.

Failed to generate these test files:
- `src/app/src/pages/eval/components/ResultsView.test.tsx`
  - `ResultsView` should render without crashing when `recentEvals` is empty and `defaultEvalId` is undefined
  - `ResultsView` should invoke `onRecentEvalSelected` with correct eval ID on recent eval selection
  - `ResultsView` should handle missing `defaultEvalId` in `recentEvals` gracefully
      
Please contact [Support](https://app.usetusk.ai/app/testing-commit-check/83892fab-e864-4984-9b83-68d122f18493?client=e91e3433-8fb4-4e98-b233-bba492f4909c&showChatBubble=true) if you have any questions.

**Comment by nidhishgajjar** [comment]
**Orb Code Review** (powered by GLM-4.7 on [Orb Cloud](https://orbcloud.dev))

## Summary

This PR adds the ability to claim evals from other users in cloud mode. Previously, users could only claim unclaimed evals. The update allows users to claim evals that were authored by other users, enabling better collaboration in shared environments. The PR also updates the AuthorChip popover to show appropriate messages depending on whether the eval is unclaimed or claimed by another user, and updates help text to reference the correct command (`promptfoo auth login` instead of `promptfoo config set email`).

## Architecture

**The feature:**

1. **Updated `canEditAuthor` logic** (`AuthorChip.tsx`):
```typescript
// Before: editable=true only when no author (claiming unclaimed)
const canEditAuthor = isCloudEnabled ? !author : true;

// After: editable=true when no author OR author is someone else (claiming from others)
const canEditAuthor = isCloudEnabled ? !author || author !== currentUserEmail : true;
```

2. **Updated AuthorChip popover messages**:
```typescript
// Before: Only claimed unclaimed evals
<Typography variant="body2">
  This eval has no author assigned.
</Typography>

// After: Show different messages based on author
<Typography variant="body2">
  {author
    ? `This eval is currently attributed to ${author}.`
    : 'This eval has no author assigned.'}
</Typography>
```

3. **Updated help text**:
```typescript
// Before:
It is changeable with `promptfoo config set email <your-email@example.com>`

// After:
It is changeable with `promptfoo auth login`
```

4. **Added tests** (`AuthorChip.test.tsx`):
```typescript
it('shows "currently attributed to" message when claiming from another user', async () => {
  render(<AuthorChip {...cloudProps} author="other@example.com" editable={true} />);
  await userEvent.click(screen.getByText('other@example.com'));
  expect(
    screen.getByText('This eval is currently attributed to other@example.com.'),
  ).toBeInTheDocument();
  expect(screen.getByRole('button', { name: /Claim as mine/i })).toBeInTheDocument();
});

it('can claim eval from another user', async () => {
  render(<AuthorChip {...cloudProps} author="other@example.com" editable={true} />);
  await userEvent.click(screen.getByText('other@example.com'));
  await userEvent.click(screen.getByRole('button', { name: /Claim as mine/i }));
  expect(mockOnEditAuthor).toHaveBeenCalledWith('user@example.com');
});
```

**Key improvements:**

1. **Better collaboration**: Users can now claim evals from other users, not just unclaimed ones

2. **Clear messaging**: Different messages for unclaimed vs claimed evals

3. **Correct help text**: References the correct command (`promptfoo auth login`)

4. **Comprehensive tests**: Tests for claiming from other users

## Issues

**No issues found.**

This is a well-implemented feature that:

1. **Enables better collaboration**: Users can claim evals from other users in shared environments

2. **Clear UX**: Different messages help users understand what they're claiming

3. **Correct documentation**: Updated help text to reference the correct command

4. **Good tests**: Tests verify the new claiming behavior

5. **Backward compatible**: Existing functionality for claiming unclaimed evals still works

**Minor suggestion:**

Consider adding a test that verifies unclaimed eval claiming still works (to ensure backward compatibility wasn't broken).

## Cross-file Impact

**Affected components:**
- `src/app/src/pages/eval/components/AuthorChip.tsx` — Updated canEditAuthor logic and messages (10 insertions, 4 deletions)
- `src/app/src/pages/eval/components/AuthorChip.test.tsx` — Added tests for claiming from others (22 insertions)
- `src/app/src/pages/eval/components/ResultsView.tsx` — Updated for new behavior (19 insertions, 9 deletions)
- `src/app/src/pages/eval/components/ResultsView.test.tsx` — Updated tests (5 insertions)

**Breaking changes:** None. This is a feature enhancement.

**Behavior changes:**
- **Before**: Users could only claim unclaimed evals in cloud mode
- **After**: Users can claim evals from other users in cloud mode (enables collaboration)
- **Before**: Help text referenced `promptfoo config set email`
- **After**: Help text references `promptfoo auth login`

**Cloud mode impact:**
- **Positive**: Enables better collaboration in shared cloud environments
- **Use case**: Team members can claim and work on each other's evals
- **Permissions**: Still requires appropriate permissions to edit eval author

## Assessment

✅ Approve

Well-implemented feature that improves cloud mode collaboration:

**Strengths:**
- Enables better collaboration by allowing claims from other users
- Clear UX with different messages for unclaimed vs claimed evals
- Correct help text pointing to the right command
- Tests verify the new claiming behavior
- Backward compatible with existing unclaimed claiming

**User benefit:**
Team members working in shared cloud environments can now claim and work on each other's evals, enabling better collaboration and handoff. The clear messaging helps users understand what they're claiming (unclaimed vs claimed by another user).

The minor suggestion about testing unclaimed claiming is for ensuring backward compatibility but not a blocker for merging.

**Comment by mldangelo-oai** [comment]
@codex review

**Comment by mldangelo-oai** [comment]
Updated this PR with current `main` and did a fresh review/verification pass on head `61fb429ae1d586ac36eedfb9bee3001b08b23b9e`.

What I checked:
- Confirmed the historical review threads are resolved/outdated on the current diff.
- Reviewed the cloud author ownership path end to end: unassigned cloud evals can only be claimed by the authenticated email, assigned cloud evals become immutable, and self-hosted editing remains available when cloud is disabled.
- Ran focused app tests: `npm run test:app -- src/pages/eval/components/AuthorChip.test.tsx src/pages/eval/components/EvalHeader.test.tsx src/pages/eval/components/ResultsView.test.tsx src/utils/api.test.ts --run` (4 files, 100 tests passed).
- Ran focused server/redteam-adjacent tests: `npx vitest run test/server/eval.test.ts test/server/routes/eval.validation.test.ts test/server/routes/user.test.ts test/redteam/extraction/entities.test.ts test/redteam/extraction/purpose.test.ts test/redteam/extraction/util.test.ts --sequence.shuffle=false` (6 files, 96 tests passed).
- Ran `npm run l`, `git diff --check`, `npm run tsc`, and `npm run build` successfully. `npm run f` hit the repo's changed-file formatter edge case after reporting the existing `src/models/eval.ts` complexity warning and then Prettier received no files; no formatting changes were needed.
- Ran an isolated E2E smoke with a temp `PROMPTFOO_CONFIG_DIR`/cache: created a real eval via `npm run local -- eval`, started `npm run dev:server` on the temp DB with cloud enabled, verified `/api/user/email` and `/api/user/cloud-config`, confirmed a wrong-email claim gets 403, a current-user claim gets 200, and a second author change gets 403. Verified the temp DB stored the claimed author, then stopped the server and removed the temp DB/cache.

CI has started on the pushed head; moving on per the sweep instruction rather than waiting for all checks here.

File: src/models/eval.ts [hunk 1]
```diff
@@ -665,7 +684,8 @@ export default class Eval {
       isRedteam: this.config.redteam !== undefined,
       prompts: this.prompts,
       description: this.config.description,
-      author: this.author,
+      // Use null explicitly to ensure Drizzle updates the column (undefined may be skipped)
+      author: this.author || null,
```
Comment (jameshiester-oai) [inline src/models/eval.ts:688]: **P1 — Keep general saves from overwriting the authoritative author claim.** `updateAuthor` now writes the claim atomically, but every later `save()` writes the cached `this.author`. I reproduced this by retaining an instance with `author === null`, successfully calling `Eval.updateAuthor(id, "cloud-user@example.com", { onlyIfUnassigned: true })`, and then calling `save()` on the retained instance: the database author reverted to null. In-flight evaluations and the rating path can therefore silently undo a claim and reopen it. Remove `author` from the general save payload (or guard it with an expected-current-value check), route intentional author writes through the dedicated operation, and add this interleaving as a regression test.

**Review by jameshiester-oai — COMMENTED** [review]
## Review summary

Reviewed base `c54a30668ad8319d76c20ae96e6680ad6c51a2c6` → head `840b648dcebbf48e105f30dbfbdd9f849e47ae21`.

**Scope:** full current PR. No prior authenticated exact-head completion marker exists, so this was not an incremental review.

**Verdict:** the current head has one P1 data-integrity defect and four P2 correctness/contract/CI issues. The PR title follows the repository's `(app)` convention. GitHub reports the PR mergeable but behind its base.

**CI:** Style Check has a deterministic PR-caused architecture failure: `app -> legacy-contracts` increased from 99 to 100 and `view-server -> node` increased from 43 to 44. Site (101 tests), WebUI (4,831 tests), and Node 20 (19,686 passed, 10 skipped) completed their test phases successfully, then failed only in Codecov's GPG step with `No public key`; I classify those as upload/infrastructure failures rather than test regressions. The remaining reported checks are green.

### P1

- **A stale `Eval` instance can erase an atomically claimed Cloud author.** `Eval.updateAuthor(..., { onlyIfUnassigned: true })` writes the claim directly, but general `Eval.save()` still writes its cached `this.author`. I reproduced the interleaving: create/load an eval with a null author, claim it successfully, then save the old instance; the database author changes from `cloud-user@example.com` back to null. Long-running evaluation and rating paths call this general save, so unrelated persistence can silently reopen ownership.

### P2

- **Cloud identity validation can commit a stale login after credentials change.** A deferred validation for token/host A returned after switching the active token, host, and stored email to B; `getCloudUserEmail()` returned A and persisted email A while token/host B remained active. A concurrent claim can therefore be locked to the wrong account.
- **A transient `/api/user/email` failure strands the claim UI.** `fetchUserEmail` maps HTTP/network/decoding failures to null, the header fetches once, and null disables the real `AuthorChip` action. There is no retry or surfaced error, so the user cannot claim until remount/reload.
- **The PR currently fails the repository architecture gate.** The new `@promptfoo/types` import in the app test creates the extra `app -> legacy-contracts` edge, and the direct `globalConfig/cloud` route import creates the extra `view-server -> node` edge. Both should be removed through an existing/narrower boundary rather than accepted into the baseline.
- **OpenAPI does not describe the new normal 403 responses.** `src/openapi/server.ts:489-504` and `:1087-1112` omit the Cloud claim/edit 403 cases now returned by the eval and user routes. Generated clients therefore cannot model an expected result of the new ownership contract.

### Existing unresolved feedback

- [The existing Cloud-creation identity thread](https://github.com/promptfoo/promptfoo/pull/6759#discussion_r3368894459) still applies; I did not duplicate it inline. The end-to-end probe also confirmed that v4 `POST /api/eval` accepts an explicit arbitrary `author`, persists it in Cloud mode, and then rejects repair by the verified token owner. Creation/import provenance needs to be separated from authenticated Cloud ownership.
- [The existing mock-isolation thread](https://github.com/promptfoo/promptfoo/pull/6759#discussion_r3368894462) remains current for the three extraction test files; I did not duplicate it.

### Simplification and coverage opportunities

Use one cached/synchronized Cloud identity source (the existing user context/store is a natural UI boundary) instead of independently validating from the header and root provider. On the server, a dedicated bodyless Cloud claim operation would avoid conflating authenticated ownership with the self-hosted arbitrary-author edit API. Add interleaving tests for claim-versus-save and credential-switch-during-validation, plus a failed-first/retry UI test using the real `AuthorChip` composition.

### Validation performed

All commands ran on the exact detached head with Node 24.16.0 and npm 11.13.0:

- `npx vitest --run test/globalConfig/accounts.test.ts test/models/eval.test.ts test/server/eval.test.ts test/server/routes/eval.validation.test.ts test/server/routes/user.test.ts test/redteam/extraction/entities.test.ts test/redteam/extraction/purpose.test.ts test/redteam/extraction/util.test.ts` — exit 0; 8 files, 299 tests passed.
- `npm run test:app -- src/pages/eval/components/AuthorChip.test.tsx src/pages/eval/components/EvalHeader.test.tsx src/pages/eval/components/ResultsView.test.tsx src/utils/api.test.ts --run` — exit 0; 4 files, 102 tests passed.
- `npx vitest --run test/openapi/serverOpenApi.test.ts test/server/routes/serverRouteSmoke.test.ts` — exit 0; 2 files, 72 tests passed.
- `npm run tsc`, `npm run build`, `npm run lint:src`, and `npm run format:check` — exit 0. Lint/format emitted warning-only pre-existing complexity findings.
- `npm run architecture:check` — exit 1 with the two exact edge regressions described above.
- `npx tsx /private/tmp/pr6759-cloud-create-probe.ts` under a scrubbed environment and isolated config/database/cache — exit 0. Verified identity GET 200, wrong claim 403, correct unassigned claim 200, Cloud clear 403, self-hosted set/clear 200, and reproduced explicit Cloud creation with a spoofed author followed by repair 403.
- `npx tsx /private/tmp/pr6759-stale-save-probe.ts` with an isolated database — exit 0; reproduced claim success followed by stale-save reset to null.
- `npx tsx /private/tmp/pr6759-token-race-probe.ts` with isolated global config — exit 0; reproduced token B/host B paired with returned and persisted email A.
- Worktree remained clean after validation.

### Limitations

The requested `/pr-review-toolkit:review-pr all parallel` command is unavailable in this environment (exit 127), so I performed the closest read-only six-lens review manually and reconciled three independent read-only specialist reports. I did not use live Promptfoo Cloud credentials or paid model calls; Cloud behavior was exercised against a loopback fake token-validation service. I did not run browser QA because port 3000 is occupied by an unrelated forwarded application, and I did not replace it. I did not rerun the complete local test matrix; the focused suites above and the CI test phases provide broader coverage.

File: src/globalConfig/accounts.ts [hunk 2]
```diff
@@ -77,6 +77,22 @@ export function getUserEmail(): string | null {
...
+export async function getCloudUserEmail(): Promise<string> {
+  const apiKey = cloudConfig.getApiKey();
+  if (!apiKey) {
+    throw new Error('Cloud API key is not configured');
+  }
+
+  const { user } = await cloudConfig.validateApiToken(apiKey, cloudConfig.getApiHost());
+  const email = z.email().parse(user.email);
+
+  if (getUserEmail() !== email) {
```
Comment (jameshiester-oai) [inline src/globalConfig/accounts.ts:89]: **P2 — Reject validation results from superseded credentials.** This async operation validates the token/host captured at call time but commits its email without checking that those credentials are still active. In a deferred-validation probe, I switched token, host, and email from A to B before A resolved; this function returned and persisted email A while token/host B remained active. A concurrent claim can then be assigned to the wrong account. Capture both token and host before awaiting, compare them with the current values after validation, and abort or retry if either changed.

File: src/app/src/pages/eval/components/EvalHeader.tsx [hunk 3]
```diff
@@ -64,6 +65,11 @@ export default function EvalHeader({
...
   const [evalActionsOpen, setEvalActionsOpen] = React.useState(false);
   const [currentUserEmail, setCurrentUserEmail] = React.useState<string | null>(null);
+  const {
+    data: cloudConfig,
+    isLoading: isCloudConfigLoading,
+    error: cloudConfigError,
+  } = useCloudConfig();
 
   React.useEffect(() => {
     fetchUserEmail().then((email) => {
```
Comment (jameshiester-oai) [inline src/app/src/pages/eval/components/EvalHeader.tsx:75]: **P2 — Provide an error/retry path for Cloud identity loading.** `fetchUserEmail` collapses HTTP, network, and response-decoding failures to `null`, and this one-shot effect never retries or records an error. The resulting null makes `canEditAuthor` false, so the real `AuthorChip` cannot even open its loading dialog; a transient failure disables claiming until remount/reload. Track loading/error/value separately and expose a retry (ideally through the shared user identity context), with a failed-first/retry composition test.

File: src/app/src/pages/eval/components/EvalHeader.test.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
+import { renderWithProviders } from '@app/utils/testutils';
+import { waitFor } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import EvalHeader from './EvalHeader';
+import { useTableStore } from './store';
+import type { ResultLightweightWithLabel } from '@promptfoo/types';
```
Comment (jameshiester-oai) [inline src/app/src/pages/eval/components/EvalHeader.test.tsx:7]: **P2 — Remove the new forbidden app-to-legacy-contracts edge.** The repository's Style Check reports `app -> legacy-contracts: 100 imports (was 99)`, and this new direct `@promptfoo/types` import is the added edge. Infer the fixture type from the component/store API or use the approved app-local contract boundary; increasing the architecture baseline would preserve the regression.

File: src/server/routes/eval.ts [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import { Router } from 'express';
 import { z } from 'zod';
 import { HUMAN_ASSERTION_TYPE } from '../../constants';
-import { getUserEmail, setUserEmail } from '../../globalConfig/accounts';
+import { getCloudUserEmail, getUserEmail, setUserEmail } from '../../globalConfig/accounts';
+import { cloudConfig } from '../../globalConfig/cloud';
```
Comment (jameshiester-oai) [inline src/server/routes/eval.ts:5]: **P2 — Keep the route behind the approved server boundary.** The repository's Style Check reports `view-server -> node: 44 imports (was 43)`; this direct `globalConfig/cloud` import is the added edge. Expose the narrow Cloud-enabled/identity decision through the already-used account boundary (or another approved server-facing abstraction) so this route does not add a node-layer dependency. Do not raise the architecture baseline for this single check.
